src/client/GroupCircle.hpp

```diff
@@ -0,0 +1,21 @@
+#ifndef GroupCircle_hpp
+#define GroupCircle_hpp
+
+#include <SFML/Graphics.hpp>
+#include <stdio.h>
+#include "../common/Circle.hpp"
+#include "../common/GameObject.hpp"
+
+class GroupCircle : public GameObject {
```

Comment:
Niiiiice, I way prefer this way. One thing Im a little confused about is the naming here. Should this just be called Group?  I'm actually not sure why it exists also. Is it just to hold the group ID? if it's going to be the Group class, Im guessing down the line it can hold a list of member clients?

Reply:
This is the representation of a Group for the client - it's just a circle representing a group. 
I think we should leave the design this way for now since it's unclear how much the client will actually need to know in the future.
With this we can do velocity and collision interpolation on the client.

If you can think of a better name, I'm down to change it.

Reply:
The actual `Group` class is in the server (`Group.hpp`)

Reply:
I know Im annoying, but my fundamental question is: why aren't the group representations pretty much the same on server and client? Why aren't both of these called Group?

Reply:
Because the client doesn't need all the functionality of `Group`, at least not for now. We want to avoid sending all the data and writing all the code needed to update/reconstruct a group.

The representation of a Group on the client will be clearer as we go along, but for the foreseeable future all we need is circle that we can interpolate movement and collision for.

imo

Reply:
Aha, then maybe we shouldn't be sending groups, we should be sending circles?

Reply:
Well the client needs to know that it's receiving a Group not just a circle.
What data we send exactly is a discussion for https://github.com/SourenP/ungroup_sfml/pull/19

Would like to figure out how we want clients to interpret/represent Groups.

Reply:
sounds good, so we basically are just calling this GroupCircle to avoid a naming collision with the name Group on server side, but they're basically the same object?